### PR TITLE
LOG-2220: Fluentd collector not setting labels from /var/log/pods paths

### PR DESCRIPTION
**Pull request by alanconway**
Collector plugin was using very old version of prometheus-client that did not use keyword arguments, so arguments were treated as labels.  Updated the dependencies.

/cc @jcantrill

### CODE REVIEW

File: fluentd/lib/fluent-plugin-collected/Gemfile.lock [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
-    concurrent-ruby (1.1.9)
+    concurrent-ruby (1.1.10)
     cool.io (1.7.1)
-    fluent-plugin-prometheus (1.8.5)
+    fluent-plugin-prometheus (2.0.2)
       fluentd (>= 1.9.1, < 2)
-      prometheus-client (< 0.10)
-    fluentd (1.14.4)
+      prometheus-client (>= 2.1.0)
+    fluentd (1.14.6)
```
Comment (jcantrill) [inline fluentd/lib/fluent-plugin-collected/Gemfile.lock:16]: We are using 1.14.5 and should lock it in the gemspec or Gemfile

File: fluentd/lib/fluent-plugin-collected/fluent-plugin-collected.gemspec [hunk 2]
```diff
@@ -13,12 +13,10 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "fluentd", ">= 0.14.20", "< 2"
-  spec.add_dependency "prometheus-client", "< 0.10"
+  spec.add_dependency "fluentd", ">= 0.14.5", "< 2"
```
Comment (jcantrill) [inline fluentd/lib/fluent-plugin-collected/fluent-plugin-collected.gemspec:16]: I don't know how this bubbles up to the vendoring but we should ensure we only use 1.14.5

File: fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb [hunk 3]
```diff
@@ -9,28 +9,15 @@ class CollectedTailMonitorInput < Fluent::Plugin::Input
...
 
-    MONITOR_IVARS = [
-      :tails,
-    ]
-
-    REGEX_VAR_LOG_PODS = Regexp.new('/var/log/pods/(?<namespace>[a-z0-9-]+)_(?<pod_name>[a-z0-9-]+)_(?<pod_uuid>[a-z0-9-]+)/(?<container_name>[a-z0-9-]+)/.*\.log$')
-    REGEX_VAR_LOG_CONTAINERS = Regexp.new('/var/log/containers/(?<pod_name>[a-z0-9-]+)_(?<namespace>[a-z0-9-]+)_(?<container_name>[a-z0-9-]+)-(?<docker_id>[a-z0-9]{64})\.log$')
-    REGEX_LOG_PATH=/(#{REGEX_VAR_LOG_PODS})|(#{REGEX_VAR_LOG_CONTAINERS})/
+    MONITOR_IVARS = [:tails].freeze
+    REGEX_LOG_PATH = Regexp.new('.*/(?<namespace>[a-z0-9-]+)_(?<pod_name>[a-z0-9-]+)_(?<pod_uuid>[a-z0-9-]+)/(?<container_name>[a-z0-9-]+)/.*\.log$').freeze
```
Comment (jcantrill) [inline fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb:16]: This blocks us from possibly pushing this into older releases.  Are you sure we will never push this version of the image into logging releases older then 5.4

Comment (alanconway) [reply fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb:16]: Do you mean you want to re-enable support for /var/log/containers?

File: fluentd/lib/fluent-plugin-collected/Gemfile.lock [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
 
 GEM
   remote: https://rubygems.org/
   specs:
-    concurrent-ruby (1.1.9)
+    concurrent-ruby (1.1.10)
     cool.io (1.7.1)
-    fluent-plugin-prometheus (1.8.5)
+    fluent-plugin-prometheus (2.0.2)
       fluentd (>= 1.9.1, < 2)
```
Comment (cahartma) [inline fluentd/lib/fluent-plugin-collected/Gemfile.lock:14]: This plugin is updated, can we also set fluentd to 1.14.5 here?

Comment (jcantrill) [reply fluentd/lib/fluent-plugin-collected/Gemfile.lock:14]: wont matter her as the lock pegs it to 1.14.5 later in the lock file

File: fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb (hunk 3 above)
Comment (jcantrill) [reply fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb:16]: I'm suggesting that if we do not support /var/log/containers then this blocks us from taking fluent 1.14.5 into previous releases of logging that are still using /var/log/containers.  If we truly have no intentions of moving 1.14.5 into anything earlier then 5.4 then no need to alter what is happening here

File: fluentd/lib/fluent-plugin-collected/Gemfile.lock (hunk 4 above)
Comment (alanconway) [reply fluentd/lib/fluent-plugin-collected/Gemfile.lock:14]: I did set fluentd to 1.14.5 everywhere just to be clear.

File: fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb (hunk 3 above)
Comment (alanconway) [reply fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb:16]: Understood, I will restore support for /var/log/containers. I wanted to simplify but did not think that through.

**Pull request merged**